estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,54 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+class estate_property_offer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'property offer'
+    _sql_constraints = [
+        ('offer_price_positive', 'CHECK(price >= 0)', 'An offer price must be strictly positive'),
+    ]
+    _order = 'price desc'
+
+    price = fields.Float(string = 'price')
+    status = fields.Selection([('accepted', 'Accepted'),('refused', 'Refused')], string = 'status')
+    partner_id = fields.Many2one('res.partner', string='Partner')
+    property_id = fields.Many2one('estate.property', required=True)
+    validity = fields.Integer(default = 7, string = "Validity (days)")
+    date_deadline = fields.Date(string = "Deadline", compute = '_compute_date_deadline', inverse='_inverse_date_deadline',  store = True)
+    property_type_id = fields.Many2one(related="property_id.property_type_id", string="Property Type", store=True)
+
+    @api.depends('validity')
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = fields.Date.today() + relativedelta(days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            if record.date_deadline:
+                delta = record.date_deadline - fields.date.today()
```

Comment:
nitpick: 
```suggestion
                delta = record.date_deadline - fields.Date.today()
```
note: yours works without traceback because `fields` happens to import `date` from `datetime` and the `Date.today` wrapper doesn't actually add any logic